apps/speed.c

```diff
@@ -332,7 +332,7 @@ static double Time_F(int s)
         if (thr == NULL) {
             DWORD ret = GetLastError();
             BIO_printf(bio_err, "unable to CreateThread (%d)", ret);
-            TerminateProcess(GetCurrentProcess(), err);
+            TerminateProcess(GetCurrentProcess(), ret);
```

Comment:
This code looks wrong. TerminateProcess does not *immediately* terminate the process - it is asynchronous. Therefore this code can continue and call CloseHandle with a NULL pointer. According to the documentation we need to call WaitForSingleObject() to actually then wait for the process to exit. The same applies to the TerminateProcess() call added to uplink.

If we need a quick fix because of a compilation error I suggest reverting the previous commit.